### PR TITLE
Features MIDI file

### PR DESCRIPTION
**Pull request by antoinelefebvre17**

### CODE REVIEW

File: Features/FeaturesMIDI.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # 4: end
+    def getTemposAverageByParts(self):
+        averageMIDI = []
+        temposSplit5 = np.array_split(self.tempos, 5)
+        for i in range(len(temposSplit5)):
+            average = 0
+            for j in range(len(temposSplit5[i])):
+                average = average + temposSplit5[i][j]
+            averageMIDI.append(round(average / len(temposSplit5[i]), 2))
+        return averageMIDI
```
Comment (armandmgt) [inline Features/FeaturesMIDI.py:34]: Ça ne prend pas en compte le timecode auquel le changement de tempo a eu lieu ce code ?
Je pense que la donnée est intéressante si la moyenne est prise par rapport au moment dans la chanson

**Review by armandmgt — COMMENTED** [review]
Nice work 👍  Peut-être qu'on peut rename la PR en "MIDI Files – Tempo features" ?

File: Features/DataMIDI.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            for msg in track:
+                if msg.type == 'set_tempo':
+                    # fill tempo
+                    self.features.tempos.append(round(tempo2bpm(msg.tempo)))
+                    # fill tempo with repetition
+                    find = False
+                    for j in range(len(self.features.temposWithRepetition)):
+                        if self.features.temposWithRepetition[j].get('tempo') == round(tempo2bpm(msg.tempo)):
+                            self.features.temposWithRepetition[j]['repetition'] = self.features.temposWithRepetition[j][
+                                                                                      'repetition'] + 1
```
Comment (zombre98) [inline Features/DataMIDI.py:25]: It's cool but then you extract a note counter and if we end up with longer music we'll get a lot more notes. So the number of notes will depend a lot on the size of the music. I think you have to divide each repetition by the total number of notes to get a percentage. And rename the counter as an average. WDYT?

**Review by zombre98 — CHANGES_REQUESTED** [review]

File: Features/DataMIDI.py (hunk 2 above)
Comment (armandmgt) [reply Features/DataMIDI.py:25]: Careful, this is not about notes, it's about `tempo` MIDI messages. The point is still valid but we should take into account the time in this computation.
If we want to get `percentage of time where the tempo was n BPM`, the time must be taken into account.

**Pull request closed without merging**